lib/util/glob-util.js

```diff
@@ -43,9 +43,7 @@ function processPath(options) {
     const cwd = (options && options.cwd) || process.cwd();
     let extensions = (options && options.extensions) || [".js"];
 
-    extensions = extensions.map(function(ext) {
-        return ext.charAt(0) === "." ? ext.substr(1) : ext;
-    });
+    extensions = extensions.map(ext => ext.replace(/^\./, ""));
```

Comment:
I modified the logic here to prevent `no-confusing-arrow` from reporting an error.